source/misc/bash_startup.in

```diff
@@ -1,257 +1,95 @@
-#!/bin/bash
-# This is based on "preexec.bash" but is customized for iTerm2.
+# This is inspired by "preexec.bash" but is customized for iTerm2. Sourced, so
+# no shebang. This module requires 2 bash features you must not otherwise be
+# using: the DEBUG trap and the PROMPT_COMMAND variable.
 
-# Note: this module requires 2 bash features which you must not otherwise be
-# using: the "DEBUG" trap, and the "PROMPT_COMMAND" variable.  iterm2_preexec_install
-# will override these and if you override one or the other this _will_ break.
+osc='\033]'; cso='\007' # Beginning, ending escape sequences.
 
-# This is known to support bash3, as well as *mostly* support bash2.05b.  It
-# has been tested with the default shells on MacOS X 10.4 "Tiger", Ubuntu 5.10
-# "Breezy Badger", Ubuntu 6.06 "Dapper Drake", and Ubuntu 6.10 "Edgy Eft".
+# tmux / screen are not supported; I think this is the theoretical tmux hack?
+[ $TERM = screen ] && osc='\033Ptmux;\033\033]' && cso='\007\033\\'
```

Comment:
FYI, It was <a href="https://gitlab.com/gnachman/iterm2/issues/3118#note_1547513">suggested to me</a> that this would be a better test for TERM:

````["${TERM:-}" != screen*]````
This would seem to handle an empty $TERM better, but I'm not sure what benefit the * confers.